### ISSUE TITLE
Reopen / comment on #5043 (Add .json extension is missing from webpack resolve.extensions)

### ISSUE THREAD

**Issue by broofa** [angular/angular-cli#13576]
See #5043  

tl;dr Please reopen that issue (and close this one)

Opening a new issue here because 
1. [This comment](https://github.com/angular/angular-cli/issues/5043#issuecomment-401926712) has gone unresponded to for 6 months
1. Tried reaching out to https://gitter.im/angular/angular-cli with no success
1. I just pinged @filipesilva on that issue, but I'm a little low on blood sugar (read "impatient") and kind of annoyed that I find myself having to yet again deal with an Angular user who expects me to fix a problem that pretty clearly seems to be an Angular issue.

So...  Sorry?  I don't want to be an ass by intentionally opening a duplicate issue, but it'd be nice to see some movement on this, or at least an explanation for not moving on it.

**Comment by clydin** [comment]
Don't worry about the new issue.  It's actually preferred to open a new issue when there is still a concern regarding a long closed issue (in this case almost 2 years).

As to an explanation, this issue is not truly an Angular CLI specific issue.  Treating JSON as a JavaScript module is specific to the Node.js platform and is also non-standard from the viewpoint of ES modules existing in ES2015+.  If a package is to be considered cross-platform (i.e., supporting Node.js and browsers) then it should ideally not assume specific bundler configurations or bundler capabilities to workaround Node.js specific behavior.

By making a small change to line 50 (and 70-71) of the build script in the package, JSON can be completely eliminated from the situation.  This not only avoids the entire file extension concern but also provides a path to eventual ES module compliant code.
```
  fs.writeFileSync(path, 'module.exports = ' + JSON.stringify(types) + ';');
```
```
writeTypesFile(standard, path.join(__dirname, '../types', 'standard.js'));
writeTypesFile(other, path.join(__dirname, '../types', 'other.js'));
```

**Comment by alan-agius4** [comment]
Closing following the suggestions mentioned above.
